frontend/src/metabase/dashboard/context/context.tsx

```diff
@@ -42,6 +42,7 @@ export type DashboardContextOwnProps = {
   navigateToNewCardFromDashboard:
     | ((opts: NavigateToNewCardFromDashboardOpts) => void)
     | null;
+  onNewQuestion?: () => void;
```

Comment:
This is the new property in the dashboard context. We want to perform different actions between the SDK and the core dashboard.

For the core dashboard, we'll dispatch some Redux actions that change the URL, but in the SDK, we'll only modify some local component states since we can't alter users' URLs.